packages/dds/merge-tree/src/collections/list.ts

```diff
@@ -68,32 +68,60 @@ function insertAfter<T>(node: DataNode<T> | HeadNode<T>, ... items: T[]): ListNo
     });
     oldNext._prev = previousNode;
     previousNode._next = oldNext;
+    if (newRange === undefined) {
+        throw new UsageError("items must not be empty");
+    }
```

Comment:
I think this would be more readable if it directly checked for `items.length === 0` at the start of the function, and that option would also prevent bugs that could arise from the fact that we're mutating `previousNode` aka `node` before throwing this error. The mutations currently are basically a no-op, but that could change at some point.

Reply:
i did it this way as it allow me to explicitly prevent newRange from being undefined, which i never want, without casting, and without additional conditionals, as this is used in some perf critical areas. i could have just asserted, but that thrown a non-user friendly error, so i went with a more user-friendly error, which describes the only condition that could lead to this being undefined in the current code.

Reply:
added a comment